src/dataServices/dynamoDbBundleServiceHelper.ts

```diff
@@ -253,4 +256,211 @@ export default class DynamoDbBundleServiceHelper {
 
         return { stagingResponse, itemLocked };
     }
+
+    static async sortBatchRequests(
+        requests: BatchReadWriteRequest[],
+        dynamoDbHelper: DynamoDbHelper,
+        tenantId?: string,
+    ) {
+        const deleteRequests: any[] = [];
+        const createRequests: any[] = [];
+        const updateRequests: any[] = [];
+
+        const batchReadWriteResponses: BatchReadWriteResponse[] = [];
+        // eslint-disable-next-line no-restricted-syntax
+        for (const request of requests) {
+            let vid = 0;
+            let { id } = request;
+            const { resourceType, operation } = request;
+            let item;
+            // we need to query to get the VersionID of the resource for non-create operations
+            if (operation === 'create') {
+                vid = 1;
+                id = request.id ? request.id : uuidv4();
+            } else {
+                try {
+                    // eslint-disable-next-line no-await-in-loop
+                    item = await dynamoDbHelper.getMostRecentUserReadableResource(resourceType, id, tenantId);
+                    vid = Number(item.resource?.meta.versionId);
+                } catch (e: any) {
+                    console.log(`Failed to find resource ${id}`);
+                    batchReadWriteResponses.push({
+                        id,
+                        vid: '',
+                        operation,
+                        resourceType,
+                        resource: {},
+                        lastModified: '',
+                    });
+                }
+            }
+            switch (operation) {
+                case 'create': {
+                    item = DynamoDbUtil.prepItemForDdbInsert(
+                        request.resource,
+                        id,
+                        vid,
+                        DOCUMENT_STATUS.AVAILABLE,
+                        tenantId,
+                    );
+
+                    createRequests.push({
+                        Statement: `INSERT INTO "${RESOURCE_TABLE}" VALUE ${this.convertToPartiQLString(item)}`,
+                        id,
+                    });
+
+                    batchReadWriteResponses.push({
+                        id,
+                        vid: item.meta.versionId,
+                        operation: request.operation,
+                        lastModified: item.meta.lastUpdated,
+                        resourceType,
+                        resource: item,
+                    });
+                    break;
+                }
+                case 'update': {
+                    if (vid === 0) {
```

Comment:
Does vid === 0 means it's deleted? If so, an error result should still be returned for that request.

Reply:
vid === 0 means that the resource was not found, and that we already appended the error to `batchReadWriteResponses`, so we don't have to continue there. Unfortunately using `continue` or `break` in the earlier if statement caused unexpected behavior with regards to the asynchronous behavior of the for loop.